collector/lib/ConnTracker.cpp

```diff
@@ -40,6 +40,24 @@ static const NRadixTree private_networks_tree(PrivateNetworks());
 
 }  // namespace
 
+bool AdvertisedEndpointEquality::operator()(const ContainerEndpoint& lhs, const ContainerEndpoint& rhs) const {
```

Comment:
Perhaps there should be unit tests that tests this operator directly.

Reply:
Write a comment in the current comparison operator for container endpoints pointing out that it is only comparing the addresses.

Reply:
I added a unit test targeting specifically the comparator, good idea.
Added a comment in the operator==() of endpoints to explain that using pointer comparison is a perf trick; and is currently possible thanks to guaranties provided by ProcessStore.

Reply:
Great. I did add a correction to the comment.